### PR TITLE
Console: Initial integration

### PR DESCRIPTION
**Pull request by medikoo**
Brings in initial console integration.

Depends on:
- https://github.com/serverless/dashboard-plugin/pull/674
- `@serverless/aws-lambda-otel-extension-dist` to be published (happening shortly in context of https://github.com/serverless/runtime)

I do not bring any user documentation at this point. I assume it'll be provided separately

### CODE REVIEW

File: lib/classes/plugin-manager.js [hunk 1]
```diff
@@ -519,7 +522,16 @@ class PluginManager {
   }
 
   async runHooks(hookName, hooks) {
-    for (const { hook } of hooks) await hook();
+    const debugLog = log.get('lifecycle:command:invoke:hook');
```
Comment (pgrzesik) [inline lib/classes/plugin-manager.js:525]: Was there a specific use case for introduction of this debug logic?

File: lib/configuration/is-dashboard-enabled.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+module.exports = ({ configuration, options }) => {
+  if (configuration) {
+    if (configuration.console && !configuration.dashboard) return false;
```
Comment (pgrzesik) [inline lib/configuration/is-dashboard-enabled.js:5]: Does that mean that you can configure both console and dashboard for the same service?

File: lib/plugins/aws/package/compile/functions.js [hunk 3]
```diff
@@ -414,6 +415,22 @@ class AwsCompileFunctions {
...
+      if (!functionResource.Properties.Environment.Variables) {
+        functionResource.Properties.Environment.Variables = {};
+      }
+      if (!functionResource.Properties.Layers) functionResource.Properties.Layers = [];
+
+      Object.assign(
+        functionResource.Properties.Environment.Variables,
+        await this.console.deferredFunctionEnvironmentVariables
+      );
+      functionResource.Properties.Layers.push({
```
Comment (pgrzesik) [inline lib/plugins/aws/package/compile/functions.js:429]: Should we validate if that layer can actually be added ? I believe there's a limit of 5 layers per function, what if someone is already using 5 of them?

File: lib/classes/console.js [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    const url = `${this.otelIngestionUrl}/org/${this.orgId}/service/${this.service}/stage/${this.stage}`;
+    const data = {
+      method: 'GET',
+      headers: {
+        'Authorization': `Bearer ${this.sdk.accessKey}`,
+        'Content-Type': 'application/json',
+      },
+    };
+    log.debug('get token: %s %o', url, data);
+    const response = await fetch(url, data);
```
Comment (pgrzesik) [inline lib/classes/console.js:95]: IS there a reason why we're doing it this way instead of having this method implemented on SDK level?

**Review by pgrzesik — COMMENTED** [review]
I have some general questions so far but overall it looks good, waiting for the final update

File: lib/classes/plugin-manager.js (hunk 1 above)
Comment (medikoo) [reply lib/classes/plugin-manager.js:525]: Yes, in v2 I was relying on [debug](https://github.com/serverless/serverless/tree/debug) branch occasionally to see what lifecycle events were triggered.

We had requested to expose that also from the community: https://github.com/serverless/serverless/pull/7658

New logging systems enables to add that naturally to the core, and as I needed that when working on the console I've included this in this PR

File: lib/configuration/is-dashboard-enabled.js (hunk 2 above)
Comment (medikoo) [reply lib/configuration/is-dashboard-enabled.js:5]: By default we turn off dashboard integration when console is on, but we want to allow users to use both, for e.g. migration purposes

Comment (pgrzesik) [reply lib/configuration/is-dashboard-enabled.js:5]: Thanks for clarification 👍

File: lib/classes/plugin-manager.js (hunk 1 above)
Comment (pgrzesik) [reply lib/classes/plugin-manager.js:525]: thanks for clarification, makes sense 👍

File: lib/classes/console.js (hunk 4 above)
Comment (medikoo) [reply lib/classes/console.js:95]: That's a good point. I'll discuss in later turn a future of SDK, and if we agree we want to stick to it I'll move it there

Comment (pgrzesik) [reply lib/classes/console.js:95]: sounds good 👍

File: lib/plugins/aws/package/compile/functions.js (hunk 3 above)
Comment (medikoo) [reply lib/plugins/aws/package/compile/functions.js:429]: Good point, I haven't thought about that. Due to limited timeframe, I won't address that now, but add it into TODO list

**Review by pgrzesik — COMMENTED** [review]
well done 👍

**Pull request merged**